onnxruntime/contrib_ops/cuda/bert/packed_attention.cc

```diff
@@ -0,0 +1,313 @@
+
+    if (relative_position_bias_dims.size() != 4) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+                             "Input 'relative_position_bias' is expected to have 4 dimensions, got ",
+                             relative_position_bias_dims.size());
+    }
+
+    if (relative_position_bias_dims[0] != batch_size) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+                             "Input 'relative_position_bias' dimension 0 should be same as batch_size, got ",
+                             relative_position_bias_dims[0]);
+    }
+
+    if (relative_position_bias_dims[1] != num_heads_) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+                             "Input 'relative_position_bias' dimension 1 should be same as number of heads, got ",
+                             relative_position_bias_dims[1]);
+    }
+
+    if (relative_position_bias_dims[2] != sequence_length) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+                             "Input 'relative_position_bias' dimension 2 should be same as sequence_length, got ",
+                             relative_position_bias_dims[2]);
+    }
+
+    if (relative_position_bias_dims[3] != sequence_length) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+                             "Input 'relative_position_bias' dimension 3 should be same as sequence_length, got ",
+                             relative_position_bias_dims[3]);
+    }
+  }
+
+  parameters.batch_size = static_cast<int>(batch_size);
+  parameters.sequence_length = static_cast<int>(sequence_length);
+  parameters.input_hidden_size = static_cast<int>(input_hidden_size);
+  parameters.hidden_size = static_cast<int>(q_hidden_size);
+  parameters.v_hidden_size = static_cast<int>(v_hidden_size);
+  parameters.head_size = static_cast<int>(q_hidden_size) / num_heads_;
+  parameters.v_head_size = static_cast<int>(v_hidden_size) / num_heads_;
+  parameters.num_heads = num_heads_;
+  parameters.scale = scale_;
+  parameters.token_count = static_cast<int32_t>(token_count);
+  parameters.has_relative_position_bias = nullptr != relative_position_bias;
+
+  return Status::OK();
+}
+
+template <typename T>
+MHARunner* PackedAttention<T>::TryGettingFusedRunner(const PackedAttentionParameters& parameters) const {
+  auto& device_prop = GetDeviceProp();
+
+  // Check whether we can use fused kernel
+  int sm = device_prop.major * 10 + device_prop.minor;
+  bool is_fMHA_supported = FusedMHARunnerFP16v2::is_supported(sm,
+                                                              parameters.head_size,
+                                                              parameters.sequence_length,
+                                                              enable_trt_flash_attention_,
+                                                              false);
+  bool use_fused_runner = !disable_fused_runner_ &&
```

Comment:
when some of these conditionals are False, we can return nullptr at the beginning of this function so that we do not need to call FusedMHARunnerFP16v2::is_supported.